src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java

```diff
@@ -0,0 +1,204 @@
+        String sPrepStmt = "select * from dateandtime where my_time = ?";
+        pstmt = connection.prepareStatement(sPrepStmt);
+        pstmt.setTime(1, TIME_TO_TEST);
+
+        rs = pstmt.executeQuery();
+        rs.next();
+        assertTrue(rs.getInt(1) == 42, "did not find correct timestamp");
+        rs.close();
+        pstmt.close();
+    }
+
+    /**
+     * Test query with date TVP
+     */
+    @Test
+    public void testQueryDateTVP() throws SQLException {
+        SQLServerDataTable tvp = new SQLServerDataTable();
+        tvp.addColumnMetadata("c1", java.sql.Types.DATE);
+        tvp.addRow(DATE_TO_TEST);
+        String sPrepStmt = "select * from dateandtime where my_date IN (select * from ?)";
+        pstmt = connection.prepareStatement(sPrepStmt);
+        ((SQLServerPreparedStatement)pstmt).setStructured(1, "dateTVP", tvp);
+
+        rs = pstmt.executeQuery();
+        rs.next();
+        assertTrue(rs.getInt(1) == 42, "did not find correct timestamp");
+        rs.close();
+        pstmt.close();
+    }
+
+    /**
+     * Test query with date TVP
+     */
+    @Test
+    public void testQueryTimestampTVP() throws SQLException {
+        SQLServerDataTable tvp = new SQLServerDataTable();
+        tvp.addColumnMetadata("c1", java.sql.Types.TIMESTAMP);
+        tvp.addRow(TIMESTAMP_TO_TEST);
+        String sPrepStmt = "select * from dateandtime where my_timestamp IN (select * from ?)";
+        pstmt = connection.prepareStatement(sPrepStmt);
+        ((SQLServerPreparedStatement)pstmt).setStructured(1, "timestampTVP", tvp);
+
+        rs = pstmt.executeQuery();
+        rs.next();
+        assertTrue(rs.getInt(1) == 42, "did not find correct timestamp");
+        rs.close();
+        pstmt.close();
+    }
+
+    /**
+     * Test query with date TVP
+     */
+    @Test
+    public void testQueryTimeTVP() throws SQLException {
+        SQLServerDataTable tvp = new SQLServerDataTable();
+        tvp.addColumnMetadata("c1", java.sql.Types.TIME);
+        tvp.addRow(TIME_TO_TEST);
+        String sPrepStmt = "select * from dateandtime where my_time IN (select * from ?)";
+        pstmt = connection.prepareStatement(sPrepStmt);
+        ((SQLServerPreparedStatement)pstmt).setStructured(1, "timeTVP", tvp);
+
+        rs = pstmt.executeQuery();
+        rs.next();
+        assertTrue(rs.getInt(1) == 42, "did not find correct timestamp");
+        rs.close();
+        pstmt.close();
+    }
+    private void createTVPs(String tvpName, String tvpType) throws SQLException {
+      stmt.executeUpdate("IF EXISTS (SELECT * FROM sys.types WHERE is_table_type = 1 AND name = '" + tvpName + "') " + " drop type " + tvpName);
+      String TVPCreateCmd = "CREATE TYPE " + tvpName + " as table (c1 " + tvpType + " null)";
+      stmt.executeUpdate(TVPCreateCmd);
+    }
+
+    @BeforeEach
+    public void testSetup() throws TestAbortedException, Exception {
+        assumeTrue(9 <= new DBConnection(connectionString).getServerVersion(),
```

Comment:
I think we can close this DBConnection by using try statement.

Reply:
Ack. (I think I've copied this line from an other test)